node/node.go

```diff
@@ -552,7 +532,24 @@ func (n *Node) GetAllESDTTokens(address string) (map[string]*esdt.ESDigitalToken
 			continue
 		}
 
-		if esdtToken.TokenMetaData != nil {
+		if core.ESDTType(esdtToken.Type) != core.Fungible {
+			if esdtToken.TokenMetaData == nil {
```

Comment:
you should check the nonce first. this is duplicated code. Please used ESDTDataStorageHandler

Reply:
done